vernac/declaremods.ml

```diff
@@ -972,6 +972,8 @@ let declare_modtype id args mtys mty_l =
   protect_summaries declare_mt
 
 let declare_include me_asts =
+  if Global.sections_are_opened () then
+    user_err Pp.(str "Modules and Module Types are not allowed inside sections.");
```

Comment:
If I may butt in, I'd vote for "Include is not allowed inside sections".

If you wonder, imagine including a module with submodules — there a user would be entitled to think the submodules are the problem.